homeassistant/components/sensor/sma.py

```diff
@@ -97,8 +99,11 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     session = async_get_clientsession(hass)
     grp = {GROUP_INSTALLER: pysma.GROUP_INSTALLER,
            GROUP_USER: pysma.GROUP_USER}[config[CONF_GROUP]]
-    sma = pysma.SMA(session, config[CONF_HOST], config[CONF_PASSWORD],
-                    group=grp)
+
+    url = "http{}://{}".format(
+        "s" if config[CONF_SSL] else "", config[CONF_HOST])
+
+    sma = pysma.SMA(session, url, config[CONF_PASSWORD], group=grp)
```

Comment:
This used to pass in host as url, now it pre-processes this. Is this a breaking change?

Reply:
Not a breaking change

Used to be IP and the lib defaulted to http. Now the lib accepts a URL as well to support https

What do we prefer in hass config?
* host (as IP/hostname) and ssl (as boolean)
* url as string

Reply:
Either is fine.

Why would you not always use SSL ?

Reply:
SSL is not supported by my model afaik

Reply:
Is there a way of detecting?